src/Traits/GenericActionsTrait.php

```diff
@@ -57,23 +57,119 @@ protected function loadFilteredChildren(string $id): array
         return $this->paginate($children->order([], true), ['order' => ['Trees.tree_left']])->toList();
     }
 
+    /**
+     * Dispatch before object load hook events.
+     *
+     * @param string $uname Object uname.
+     * @param string $type Object type.
+     * @return \Cake\Http\Response|null The resulting response of the events.
+     */
+    protected function dispatchBeforeLoadEvent(string $uname, string $type): ?Response
+    {
+        $event = $this->dispatchEvent(sprintf('Controller.beforeObjectLoad:%s', $uname), compact('uname', 'type'));
+        if ($event->getResult() !== null) {
+            return $event->getResult();
+        }
+
+        $event = $this->dispatchEvent(sprintf('Controller.beforeObjectLoad:%s', $type), compact('uname', 'type'));
+        if ($event->getResult() !== null) {
+            return $event->getResult();
+        }
+
+        $event = $this->dispatchEvent('Controller.beforeObjectLoad', compact('uname', 'type'));
+        if ($event->getResult() !== null) {
+            return $event->getResult();
+        }
```

Comment:
I don't really like the idea of dispatching three separate events, as it would be impossible to use listener priorities properly.

I'd rather dispatch just the `Controller.beforeObjectLoad` event, and then it would be the listener's responsibility to just `return;` without doing anything when `uname` and/or `type` are not relevant for them.

Reply:
Agreed.

Is it possibile to listen the event more than once in the same controller? I don't know if I prefer multiple listeners/methods or a single method with a switch.

**Option 1**

```php
public function implementedEvents()
{
    return parent::implementedEvents() + [
        'Controller.beforeObjectLoad' => 'configDocuments',
        'Controller.beforeObjectLoad' => 'configFolderPippo',
    ];
}

public function configDocuments($event, $type, $uname): void
{
    if ($type === 'documents') {
        // ...
    }
}

public function configFolderPippo($event, $type, $uname): void
{
    if ($uname === 'pippo') {
        // ...
    }
}
```

**Option 2**

```php
public function implementedEvents()
{
    return parent::implementedEvents() + [
        'Controller.beforeObjectLoad' => 'beforeObjectLoad',
    ];
}

public function beforeObjectLoad($event, $type, $uname): void
{
    switch ($uname) {
       // ...
    }
    switch ($type) {
       // ...
    }
}
```